chia/full_node/full_node_api.py

```diff
@@ -1103,15 +1107,37 @@ async def request_block_header(self, request: wallet_protocol.RequestBlockHeader
             msg = make_msg(ProtocolMessageTypes.reject_header_request, RejectHeaderRequest(request.height))
             return msg
         block: Optional[FullBlock] = await self.full_node.block_store.get_full_block(header_hash)
-        if block is not None:
-            tx_removals, tx_additions, _ = await self.full_node.blockchain.get_tx_removals_and_additions(block)
-            header_block = get_block_header(block, tx_additions, tx_removals)
-            msg = make_msg(
-                ProtocolMessageTypes.respond_block_header,
-                wallet_protocol.RespondBlockHeader(header_block),
+        if block is None:
+            return None
+
+        tx_removals: List[bytes32] = []
+        tx_additions: List[Coin] = []
+
+        if block.transactions_generator:
+
+            block_generator: Optional[BlockGenerator] = await self.full_node.blockchain.get_block_generator(block)
+            assert block_generator is not None
+
+            npc_result = await asyncio.get_running_loop().run_in_executor(
+                self.executor,
+                lambda x: get_name_puzzle_conditions(
+                    x,
+                    self.full_node.constants.MAX_BLOCK_COST_CLVM,
+                    cost_per_byte=self.full_node.constants.COST_PER_BYTE,
+                    mempool_mode=False,
+                ),
+                block_generator,
             )
-            return msg
-        return None
+
+            assert npc_result is not None
```

Comment:
When would this return `None`?

https://github.com/Chia-Network/chia-blockchain/blob/ae99b987cc93b2a59b33cfe686a9d0bddf194c92/chia/full_node/mempool_check_conditions.py#L33-L35

Reply:
mypy complained in an earlier incarnation of my patch. A more interesting question though; why isn't mypy reporting this assert as redundant?

Reply:
So there is a strict equality check option that will complain if you try to compare equality of unlike types.  I am not familiar offhand with such an option for `is not None` against a non-optional type.  You could dig into the docs and issues to see if it has been discussed though.